references/appfile/dryrun/diff_test.go

```diff
@@ -0,0 +1,109 @@
+	It("Test Live-Diff", func() {
+		readDataFromFile := func(path string) string {
+			b, err := ioutil.ReadFile(path)
+			By("Load data from " + path)
+			Expect(err).Should(BeNil())
+			return string(b)
+		}
+		By("Prepare test data")
+		appStr := readDataFromFile("./testdata/diff-input-app.yaml")
+		acStr := readDataFromFile("./testdata/diff-input-ac.yaml")
+		comp1Str := readDataFromFile("./testdata/diff-input-comp-1.yaml")
+		comp2Str := readDataFromFile("./testdata/diff-input-comp-2.yaml")
+		comp3Str := readDataFromFile("./testdata/diff-input-comp-3.yaml")
+
+		modifApp := new(v1beta1.Application)
+		modifAC := new(v1alpha2.ApplicationConfiguration)
+		modifComp1, modifComp2, modifComp3 := new(v1alpha2.Component), new(v1alpha2.Component), new(v1alpha2.Component)
+
+		err := yaml.Unmarshal([]byte(appStr), modifApp)
+		Expect(err).Should(BeNil())
+		err = yaml.Unmarshal([]byte(acStr), modifAC)
+		Expect(err).Should(BeNil())
+		err = yaml.Unmarshal([]byte(comp1Str), modifComp1)
+		Expect(err).Should(BeNil())
+		err = yaml.Unmarshal([]byte(comp2Str), modifComp2)
+		Expect(err).Should(BeNil())
+		err = yaml.Unmarshal([]byte(comp3Str), modifComp3)
+		Expect(err).Should(BeNil())
+		modifComps := []*v1alpha2.Component{modifComp1, modifComp2, modifComp3}
+
+		originalAppRev := new(v1beta1.ApplicationRevision)
+		origAppRevStr := readDataFromFile("./testdata/diff-app-rev.yaml")
+		err = yaml.Unmarshal([]byte(origAppRevStr), originalAppRev)
+		Expect(err).Should(BeNil())
+
+		By("Execute Live-diff")
+		diffOpt := &LiveDiffOption{&mockDryRun{
+			ac:    modifAC,
+			comps: modifComps,
+			err:   nil,
+		}}
+		diffResult, err := diffOpt.Diff(context.Background(), modifApp, originalAppRev)
+		Expect(err).Should(BeNil())
+		Expect(diffResult).ShouldNot(BeNil())
+
+		By("Print diff result into buffer")
+		buff := &bytes.Buffer{}
+		reportOpt := NewReportDiffOption(10, buff)
+		reportOpt.PrintDiffReport(diffResult)
+
+		By("Verify diff result")
+		diffResultStr := buff.String()
+		Expect(diffResultStr).Should(SatisfyAll(
+			ContainSubstring("Application (livediff-demo) has been modified(*)"),
+			ContainSubstring("Component (myweb-1) has been modified(*)"),
+			ContainSubstring("Component (myweb-1) / Trait (myingress/service) has been modified(*)"),
+			ContainSubstring("Component (myweb-1) / Trait (myingress/ingress) has been modified(*)"),
+			ContainSubstring("Component (myweb-1) / Trait (myscaler/scaler) has been removed(-)"),
+			ContainSubstring("Component (myweb-2) has no change"),
+			ContainSubstring("Component (myweb-2) / Trait (myingress/service) has been added(+)"),
+			ContainSubstring("Component (myweb-2) / Trait (myingress/ingress) has been added(+)"),
+			ContainSubstring("Component (myweb-3) has been added(+)"),
+			ContainSubstring("Component (myweb-3) / Trait (myingress/service) has been added(+)"),
+			ContainSubstring("Component (myweb-3) / Trait (myingress/ingress) has been added(+)"),
+		))
+
+	})
```

Comment:
Besides this test case, I hope we can add more cases:

1. live-diff with the revision just applied, so nothing difference show.
2. live-diff with added only, and the rest resources no difference.
3. live-diff with modified only, and the rest resources no difference.
4. live-diff with delete only, and the rest resources no difference.